src/crewai/memory/storage/bedrock_agentcore_storage.py

```diff
@@ -0,0 +1,498 @@
+import logging
+from datetime import datetime
+from typing import Any
+
+from botocore.client import Config
+from pydantic import BaseModel, Field, field_validator
+
+from crewai.memory.storage.interface import Storage
+
+# Setup logger
+logger = logging.getLogger(__name__)
+
+
+class BedrockAgentCoreConfig(BaseModel):
+    """Configuration for AgentCore memory storage."""
+
+    # Required fields
+    memory_id: str = Field(
+        ..., description="AWS Bedrock AgentCore Memory ID (required)"
+    )
+    actor_id: str = Field(..., description="Actor ID for the memory session (required)")
+    session_id: str = Field(
+        ..., description="Session ID for the memory session (required)"
+    )
+
+    # Optional fields with defaults
+    region_name: str = Field(
+        default="us-east-1", description="AWS region name (default: 'us-east-1')"
+    )
+
+    # Namespace configuration
+    namespaces: list[str] = Field(
+        default_factory=list,
+        description="List of namespaces to be searched (default: empty list)",
+    )
+
+    @field_validator("memory_id", "actor_id", "session_id")
+    @classmethod
+    def validate_required_ids(cls, v):
+        if not v or not v.strip():
+            raise ValueError("Required ID fields cannot be empty")
+        return v.strip()
+
+    @field_validator("region_name")
+    @classmethod
+    def validate_region(cls, v):
+        if not v or not v.strip():
+            raise ValueError("Region name cannot be empty")
+        return v.strip()
+
+    @field_validator("namespaces")
+    @classmethod
+    def validate_namespaces(cls, v):
+        if not isinstance(v, list):
+            raise ValueError("Namespaces must be a list")
+
+        validated_namespaces = []
+        for namespace in v:
+            if not namespace or not namespace.strip():
+                raise ValueError("Namespace cannot be empty")
+            validated_namespaces.append(namespace.strip())
+
+        return validated_namespaces
+
+    class Config:
+        """Pydantic configuration."""
+
+        validate_assignment = True
+        extra = "forbid"  # Prevent extra fields
+
+
+class BedrockAgentCoreStorage(Storage):
+    """
+    AWS Bedrock AgentCore Memory storage implementation for CrewAI.
+
+    This implementation provides integration with AWS Bedrock AgentCore Memory,
+    supporting long term and short term memory. For more details, see the documentation:
+    https://docs.aws.amazon.com/bedrock-agentcore/latest/devguide/memory.html.
+    """
+
+    def __init__(self, type: str, config: BedrockAgentCoreConfig):
+        super().__init__()
+
+        self._validate_type(type)
+        self.memory_type = type
+        self.config = config
+
+        # Initialize boto3 clients
+        self.memory_client: Any | None = None
```

Comment:
nit: Wouldn't `Any` also qualify `None` as a value, so an explicit `Any | None` might not be required. I am not sure if this will pass linting test as-is, but would know more after CI runs.